tests/test_core.py

```diff
@@ -1543,6 +1543,61 @@ def test_exceptions_rethrow_missing(self):
     create_file('main.cpp', 'int main() { throw; }')
     self.do_runf('main.cpp', None, assert_returncode=NON_ZERO)
 
+  def test_format_exception(self):
```

Comment:
Add `@with_both_eh_sjlj` to this test?

Reply:
Then you can also remove `DISABLE_EXCEPTION_CATCHING` below.

Reply:
It doesn't work with wasm-exceptions. I am not sure how to make it work with `wasm-exceptions`, if you like I can look into it. It would probably be entertaining. But I think it should be a separate PR.

Reply:
I see, can you open a bug about that so that we remember to get it fixed.   I imagine @aheejin will want it to work for sure.

Reply:
I opened #16380 which is where I got stuck in the wasm-exceptions case.